azurerm/resource_arm_network_profile.go

```diff
@@ -0,0 +1,381 @@
+	defer azureRMUnlockMultipleByName(vnetsToLock, virtualNetworkResourceName)
+
+	_, err = client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting Network Profile %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return err
+}
+
+func expandArmContainerNetworkInterfaceConfigurations(d *schema.ResourceData) (*[]network.ContainerNetworkInterfaceConfiguration, error) {
+	cniConfigs := d.Get("container_network_interface_configuration").([]interface{})
+	retCNIConfigs := make([]network.ContainerNetworkInterfaceConfiguration, 0)
+
+	for _, cniConfig := range cniConfigs {
+		nciData := cniConfig.(map[string]interface{})
+		nciName := nciData["name"].(string)
+		ipConfigs := nciData["ip_configuration"].([]interface{})
+
+		retIPConfigs := make([]network.IPConfigurationProfile, 0)
+		for _, ipConfig := range ipConfigs {
+			ipData := ipConfig.(map[string]interface{})
+			ipName := ipData["name"].(string)
+			subNetID := ipData["subnet_id"].(string)
+
+			retIPConfig := network.IPConfigurationProfile{
+				Name: &ipName,
+				IPConfigurationProfilePropertiesFormat: &network.IPConfigurationProfilePropertiesFormat{
+					Subnet: &network.Subnet{
+						ID: &subNetID,
+					},
+				},
+			}
+
+			retIPConfigs = append(retIPConfigs, retIPConfig)
+		}
+
+		retCNIConfig := network.ContainerNetworkInterfaceConfiguration{
+			Name: &nciName,
+			ContainerNetworkInterfaceConfigurationPropertiesFormat: &network.ContainerNetworkInterfaceConfigurationPropertiesFormat{
+				IPConfigurations: &retIPConfigs,
+			},
+		}
+
+		retCNIConfigs = append(retCNIConfigs, retCNIConfig)
+	}
+
+	return &retCNIConfigs, nil
+}
+
+func extractVnetAndSubnetNames(d *schema.ResourceData) (*[]string, *[]string, error) {
+	cniConfigs := d.Get("container_network_interface_configuration").([]interface{})
+	subnetNames := make([]string, 0)
+	vnetNames := make([]string, 0)
+
+	for _, cniConfig := range cniConfigs {
+		nciData := cniConfig.(map[string]interface{})
+		ipConfigs := nciData["ip_configuration"].([]interface{})
+
+		for _, ipConfig := range ipConfigs {
+			ipData := ipConfig.(map[string]interface{})
+			subnetID := ipData["subnet_id"].(string)
+
+			subnetResourceID, err := parseAzureResourceID(subnetID)
+			if err != nil {
+				return nil, nil, err
+			}
+
+			subnetName := subnetResourceID.Path["subnets"]
+			vnetName := subnetResourceID.Path["virtualNetworks"]
+
+			if !sliceContainsValue(subnetNames, subnetName) {
+				subnetNames = append(subnetNames, subnetName)
+			}
+
+			if !sliceContainsValue(vnetNames, vnetName) {
+				vnetNames = append(vnetNames, vnetName)
+			}
+		}
+	}
+
+	return &subnetNames, &vnetNames, nil
+}
+
+func flattenArmContainerNetworkInterfaceConfigurations(input *[]network.ContainerNetworkInterfaceConfiguration) []interface{} {
+	retCNIConfigs := make([]interface{}, 0)
+	if input == nil {
+		return retCNIConfigs
+	}
+
+	// if-continue is used to simplify the deeply nested if-else statement.
```

Comment:
I'm not sure this is a good idea, if any property is missing your skipping the element, i think we should nest if statements here so that we don't silently ignore a cfg because one property is nil.